doc/source/whatsnew/v2.1.0.rst

```diff
@@ -401,6 +401,7 @@ Groupby/resample/rolling
   the function operated on the whole index rather than each element of the index. (:issue:`51979`)
 - Bug in :meth:`DataFrameGroupBy.apply` causing an error to be raised when the input :class:`DataFrame` was subset as a :class:`DataFrame` after groupby (``[['a']]`` and not ``['a']``) and the given callable returned :class:`Series` that were not all indexed the same. (:issue:`52444`)
 - Bug in :meth:`GroupBy.groups` with a datetime key in conjunction with another key produced incorrect number of group keys (:issue:`51158`)
+- Bug in :meth:`GroupBy.quantile` may implicitly sort ``index.levels`` (:issue:`53009`)
```

Comment:
Can you add `with ``sort=False``. Also, ` replace `index.levels` doesn't seem clear to me, what do you think of replacing this with `the result index`?